source/common/config/delta_subscription_impl.h

```diff
@@ -179,19 +170,31 @@ class DeltaSubscriptionImpl : public Subscription,
     }
 
     grpc_stream_.establishNewStream();
-    subscribe(resources);
-    // The attempt stat here is maintained for the purposes of having consistency between ADS and
-    // individual DeltaSubscriptions. Since ADS is push based and muxed, the notion of an
-    // "attempt" for a given xDS API combined by ADS is not really that meaningful.
-    stats_.update_attempt_.inc();
+    updateResources(resources);
   }
 
-  void updateResources(const std::vector<std::string>& resources) override {
-    subscribe(resources);
-    stats_.update_attempt_.inc();
+private:
+  void sendDiscoveryRequest(absl::optional<UpdateAck> maybe_ack) {
```

Comment:
Probably worthy of comment in source, but just to make sure I get it. If we have a bunch of ack/nacks queued up, the very first one will have any pending subscribes/unsubscribes, and the following will be pure ack/nacks?

Reply:
Correct, and I agree; comment added.

Reply:
Correct, and I agree; comment added.